sys/linux: add description file socket_netlink_iscsi.txt

Anhang512 opened the pull request:
sys/linux/init.go: add the support of other protocol in syz_init_net_socket pseudo-syscall

*******************************************************************************
Before sending a pull request, please review Contribution Guidelines:
https://github.com/google/syzkaller/blob/master/docs/contributing.md
*******************************************************************************

Anhang512 committed:
sys/linux: add description file socket_netlink_iscsi.txt
sys/linux/init.go: add the support of other protocol in syz_init_net_socket pseudo-syscall

dvyukov commented on sys/linux/socket_netlink_iscsi.txt.const:
```diff
@@ -0,0 +1,7 @@
+# Code generated by syz-sysgen. DO NOT EDIT.
+arches = amd64
```
The const files need to be generated for all arches:

$ syz-env make extract SOURCEDIR=/src/linux

where /src/linux is the latest linux-next checkout.

dvyukov commented on sys/linux/socket_netlink_iscsi.txt:
```diff
@@ -0,0 +1,50 @@
+include <uapi/linux/netlink.h>
```
This needs the standard copyright header.

dvyukov commented on sys/linux/init.go:
```diff
@@ -234,7 +234,7 @@ func (arch *arch) neutralize(c *prog.Call, fixStructure bool) error {
 		case arch.AF_NETLINK:
 			c.Args[1].(*prog.ConstArg).Val = arch.SOCK_RAW
-			c.Args[2].(*prog.ConstArg).Val = arch.NETLINK_GENERIC
+			//c.Args[2].(*prog.ConstArg).Val = arch.NETLINK_GENERIC
```
We need to restrict protocols to NETLINK_GENERIC or NETLINK_ISCSI.

dvyukov commented on sys/linux/socket_netlink_iscsi.txt:
```diff
@@ -0,0 +1,50 @@
+include <include/linux/net.h>
+
+resource sock_nl_iscsi[sock]
+#resource session_id[int32]
```
No dead code please.

dvyukov commented on sys/linux/socket_netlink_iscsi.txt:
```diff
@@ -0,0 +1,50 @@
+}
+
+netlink_msg_create_session {
+	len		const[72, int32]
```
Can this be `len[parent, int32]`?

dvyukov commented on sys/linux/socket_netlink_iscsi.txt:
```diff
@@ -0,0 +1,50 @@
+	flags		flags[netlink_msg_flags, int16]
+	seq		netlink_seq
+	pid		netlink_port_id
+#msg head len 16
```
If we use len[parent] we don't need these hints, right?

Anhang512 committed:
sys/linux: modify description file socket_netlink_iscsi.txt
sys/linux/init.go: add the support of NETLINK_ISCSI protocol in syz_init_net_socket pseudo-syscall

Anhang512 committed:
sys/linux/init.go: add support of NETLINK_ISCSI protocol for syz_init_net_socket pseudo-syscall
